kernel/syscall.c

```diff
@@ -608,24 +617,50 @@ long myst_get_absolute_path_from_dirfd(
                 ERAISE(-ENOTDIR);
         }
 
-        /* get the full path of dirfd */
-        ECHECK((*fs->fs_realpath)(
-            fs, file, locals->dirname, sizeof(locals->dirname)));
+        if (!(buf = (char*)malloc(PATH_MAX)))
+            ERAISE(-ENOMEM);
+        // get the full path of dirfd
+        ECHECK((*fs->fs_realpath)(fs, file, buf, PATH_MAX));
```

Comment:
We don't need the temporary `buf`. We can call `fs_realpath` on `path_out` and then `strcpy` or `memcpy` the `pathname`